Tests fail on Python 3.11 due to importlib.resources DeprecationWarning

Tests fail on Python 3.11:
```
pep517/in_process/__init__.py:14: in _in_proc_script_path
    return resources.path(__package__, '_in_process.py')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = ('pep517.in_process', '_in_process.py'), kwargs = {}

    @functools.wraps(func)
    def wrapper(*args, **kwargs):
>       warnings.warn(
            f"{func.__name__} is deprecated. Use files() instead. "
            "Refer to https://importlib-resources.readthedocs.io"
            "/en/latest/using.html#migrating-from-legacy for migration advice.",
            DeprecationWarning,
            stacklevel=2,
        )
E       DeprecationWarning: path is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.

/usr/lib64/python3.11/importlib/resources/_legacy.py:18: DeprecationWarning
```

--- a/pep517/in_process/__init__.py
+++ b/pep517/in_process/__init__.py
@@ -9,10 +9,18 @@ from os.path import join as pjoin
 
 try:
     import importlib.resources as resources
-
-    def _in_proc_script_path():
-        return resources.path(__package__, '_in_process.py')
+    try:
+        resources.files
+    except AttributeError:
+        # Python 3.8 compatibility
+        def _in_proc_script_path():
+            return resources.path(__package__, '_in_process.py')
+    else:
+        def _in_proc_script_path():
+            return resources.as_file(
+                resources.files(__package__).joinpath('_in_process.py'))
 except ImportError:
+    # Python 3.6 compatibility
     @contextmanager
     def _in_proc_script_path():
         yield pjoin(dirname(abspath(__file__)), '_in_process.py')

